instrumentation/netty/netty-3.8/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/v3_8/client/NettyHttpClientTracer.java

```diff
@@ -52,7 +53,7 @@ public Context startSpan(Context parentContext, ChannelHandlerContext ctx, HttpR
   }
 
   public void connectionFailure(Context parentContext, Channel channel, Throwable throwable) {
-    SpanBuilder spanBuilder = spanBuilder(parentContext, "CONNECT", CLIENT);
+    SpanBuilder spanBuilder = spanBuilder(parentContext, "CONNECT", INTERNAL);
```

Comment:
why INTERNAL over CLIENT?

Reply:
When we always create connection span it could be nested under a client span so using `INTERNAL`. When creating connection span only for failure it could be kept as `CLIENT` though as the connection failed it shouldn't have a child `SERVER` span so changed this also to `INTERNAL`.

Reply:
As we currently suppress nested CLIENT spans, we cannot have these new spans as CLIENT. Because they will be suppressed almost always. So we should have them INTERNAL for now.

Reply:
I'd prefer to keep this one here as CLIENT, and the other one can be INTERNAL for now until we support nested CLIENT spans in #3691.

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#spankind:

> CLIENT - Indicates that the span describes a request to some remote service
> INTERNAL - Indicates that the span represents an internal operation within an application